cmd/backup-manager/app/util/remote.go

```diff
@@ -0,0 +1,249 @@
+	}
+}
+
+// getRemoteStorageForBackup returns the arg for --storage option and the remote path for br
+func getRemoteStorageForBackup(backup *v1alpha1.Backup) (string, string, error) {
+	switch backup.Spec.StorageType {
+	case "s3":
+		if backup.Spec.S3 == nil {
+			return "", "", fmt.Errorf("no s3 config in backup %s/%s", backup.Namespace, backup.Name)
+		}
+		qs := checkS3Config(backup.Spec.S3, false)
+		s, path := newS3StorageOption(qs)
+		return s, path, nil
+	default:
+		return "", "", fmt.Errorf("storage %s not support yet", backup.Spec.StorageType)
+	}
+}
+
+// getRemoteStorageForRestore returns the arg for --storage option and the remote path for br
+func getRemoteStorageForRestore(restore *v1alpha1.Restore) (string, error) {
+	switch restore.Spec.StorageType {
+	case "s3":
+		if restore.Spec.S3 == nil {
+			return "", fmt.Errorf("no s3 config in restore %s/%s", restore.Namespace, restore.Name)
+		}
+		qs := checkS3Config(restore.Spec.S3, false)
+		s, _ := newS3StorageOption(qs)
+		return s, nil
+	default:
+		return "", fmt.Errorf("storage %s not support yet", restore.Spec.StorageType)
+	}
+}
+
+// newS3StorageOption constructs the arg for --storage option and the remote path for br
+func newS3StorageOption(qs *s3Query) (string, string) {
+	path := fmt.Sprintf("s3://%s/%s", qs.bucket, qs.prefix)
+	s := fmt.Sprintf("s3://%s?%s=%s", qs.bucket, prefixKey, qs.prefix)
+	if qs.region != "" {
+		s += fmt.Sprintf("&%s=%s", regionKey, qs.region)
+	}
+	if qs.provider != "" {
+		s += fmt.Sprintf("&%s=%s", providerKey, qs.provider)
+	}
+	if qs.endpoint != "" {
+		s += fmt.Sprintf("&%s=%s", endpointKey, qs.endpoint)
+	}
+	if qs.insecure != "" {
+		s += fmt.Sprintf("&%s=%s", insecureKey, qs.insecure)
+	}
+	ak := os.Getenv(accessKeyEnv)
+	if ak != "" {
+		s += fmt.Sprintf("&%s=%s", accessKey, ak)
+	}
+	sak := os.Getenv(secretAccessKeyEnv)
+	if sak != "" {
+		s += fmt.Sprintf("&%s=%s", secretAccessKey, sak)
+	}
+
+	return s, path
+}
+
+// newS3Storage initialize a new s3 storage
+func newS3Storage(qs *s3Query) (*blob.Bucket, error) {
+	awsConfig := aws.NewConfig().WithMaxRetries(maxRetries)
+	if qs.region != "" {
+		awsConfig.WithRegion(qs.region)
+	}
+	if qs.oriEndpoint != "" {
+		awsConfig.WithEndpoint(qs.oriEndpoint)
+	}
+	forcePathStyle := true
+	if qs.provider == aliKey {
+		forcePathStyle = false
+	}
+	awsConfig.WithS3ForcePathStyle(forcePathStyle)
+	// awsConfig.WithLogLevel(aws.LogDebugWithSigning)
+	awsSessionOpts := session.Options{
+		Config: *awsConfig,
+	}
+	ses, err := session.NewSessionWithOptions(awsSessionOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a *blob.Bucket.
+	bkt, err := s3blob.OpenBucket(context.Background(), ses, qs.bucket, nil)
+	if err != nil {
+		return nil, err
+	}
+	return blob.PrefixedBucket(bkt, qs.prefix), nil
+
+}
+
+// checkS3Config constructs s3Query parameters
+func checkS3Config(s3 *v1alpha1.S3StorageProvider, fakeRegion bool) *s3Query {
+	sqs := s3Query{}
```

Comment:
there is another amazon service called "sqs". Seeing this variable name confused me at first.